hr_attendance_report_theoretical_time/models/hr_leave.py

```diff
@@ -7,6 +8,19 @@
 class HrLeave(models.Model):
     _inherit = 'hr.leave'
 
+    def recompute_leave_dates(self):
+        """Recompute leaves if resource_calendar update after create record, although
+        it is strange, it has been possible to change the work schedule"""
+        for item in self.filtered(lambda x: x.state == 'validate'):
+            item._onchange_request_parameters()
+            # change resource_calendar_leaves (remove + create)
+            item._remove_resource_leave()
```

Comment:
This seems very invasive. Is it really needed? And you should only do it for those that are not correct. You should check the initial values and compare if there's a change.